light-rest-4j/src/main/java/com/networknt/codegen/rest/OpenApiGenerator.java

```diff
@@ -273,48 +286,123 @@ public void generate(String targetPath, Object model, Any config) throws IOExcep
+                                            Any resolved = references.get(any.toString());
+                                            if (resolved == null) {
+                                                continue;
+                                            }
+                                            any = new UnresolvedTypeHolderAny(resolved);
+                                            properties.put("type", any);
+                                        }
+
+                                        do {
+                                            UnresolvedTypeAny previous = null;
+                                            while (any instanceof UnresolvedTypeAny) {
+                                                previous = (UnresolvedTypeAny)any;
+                                                any = ((UnresolvedTypeAny)any).get();
+                                            }
+
+                                            if (any == null) {
+                                                break;
+                                            }
+
+                                            if (any.valueType() == ValueType.STRING) {
+                                                any = references.get(any.toString());
+                                                if (any == null) {
+                                                    break;
+                                                } else {
+                                                    previous.set(any);
+                                                }
+                                            } else {
+                                                break;
+                                            }
+                                        } while (true);
+                                    }
+                                }
+
+                                try {
+                                    transfer(targetPath,
+                                            ("src.main.java." + modelPackage).replace(".", separator),
+                                            modelFileName + ".java",
+                                            templates.rest.pojo.template(modelPackage, modelFileName, classVarName, props));
+
+                                } catch (IOException ex) {
+                                    throw new RuntimeException(ex);
+                                }
+                            };
+                        }
+                        modelCreators.add(r);
+                    } else {
+                        HashMap<String, Any> map = new HashMap<>(1);
+                        map.put(type, Any.wrap(value));
+                        handleProperties(props, map);
+                        if (props.isEmpty()) {
+                            throw new IllegalStateException("Properties empty for " + classVarName + "!");
+                        }
+
+                        references.put(classVarName, props.get(0).get("type"));
```

Comment:
If I understand correctly, if only store type information in the refenreces, would this ignore the enum case?

For example, if the specification contains following schema
```
error:
      type: object
      properties:
        code:
          $ref: '#/components/schemas/code' 
code:
      type: string
      enum: [test, test1]
```
The code would be generated as 
```
private String code
```
However, it should be 
```
public enum CodeEnum {...}
```

Reply:
Thanks for the feedback... Part of the reason I missed that is I am not that familiar with the code. But now with your concrete example, I know what I need to review. Will fix it.

Reply:
This one is addressed too.